realm/realm-library/src/main/cpp/util.cpp

```diff
@@ -45,6 +45,10 @@ jclass java_lang_string;
 jmethodID java_lang_double_init;
 jclass java_util_date;
 jmethodID java_util_date_init;
+#if REALM_ENABLE_SYNC
+jclass java_syncmanager_class;
```

Comment:
The progress listener callback is happening on a pure C++ thread. It might not be able to lookup those classes correctly according to: https://developer.android.com/training/articles/perf-jni.html#faq_FindClass